src/axom/inlet/LuaReader.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file LuaReader.cpp
+ *
+ * \brief This file contains the class implementation of the LuaReader.
+ *******************************************************************************
+ */
+
+#include "axom/inlet/LuaReader.hpp"
+
+#include "axom/core/utilities/FileUtilities.hpp"
+#include "axom/core/utilities/StringUtilities.hpp"
+
+#include "fmt/fmt.hpp"
+#include "axom/slic.hpp"
+
+namespace axom
+{
+namespace inlet
+{
+
+LuaReader::~LuaReader()
+{
+  if (m_luaState)
+  {
+    lua_close(m_luaState);
+  }
+}
+
+
+bool LuaReader::parseFile(const std::string& filePath)
+{
+  if (!axom::utilities::filesystem::pathExists(filePath))
+  {
+    SLIC_WARNING(fmt::format("Inlet: Given Lua input deck does not exist: {0}",
+                             filePath));
+    return false;
+  }
+
+  m_luaState = luaL_newstate();
+  if (luaL_loadfile(m_luaState, filePath.c_str()) ||
+      lua_pcall(m_luaState, 0, 0, 0))
+  {
+    SLIC_WARNING(fmt::format(
+                   "Inlet: Given Lua input deck could not be loaded: {0}",
+                   filePath));
+    m_luaState = nullptr;
+    return false;
+  }
+
+  return true;
+}
+
+
+bool LuaReader::parseString(const std::string& luaString)
+{
+  if (luaString.empty())
+  {
+    SLIC_WARNING("Inlet: Given an empty Lua string to parse.");
+    return false;
+  }
+
+  m_luaState = luaL_newstate();
+  if (luaL_loadstring(m_luaState, luaString.c_str()) ||
+      lua_pcall(m_luaState, 0, 0, 0))
+  {
+    SLIC_WARNING(fmt::format("Inlet: Given Lua string could not be loaded: {0}",
+                             luaString));
+    m_luaState = nullptr;
+    return false;
+  }
+
+  return true;
+}
+
+// TODO allow alternate delimiter at sidre level
+#define SCOPE_DELIMITER '/'
+
+bool LuaReader::findVariable(const std::string& id)
```

Comment:
suggestion:  Rename to `hasVariable` ?  
`findVariable` makes me think that it will return something, e.g. a pointer to the found item.

Reply:
This does more than just confirm that it is there.  It moves the Lua state to where the variable is so it can then be accessed/used.